packages/calling-stateful-client/src/CallAgentDeclarativeCommon.ts

```diff
@@ -0,0 +1,287 @@
+        } else {
+          /* @conditional-compile-remove(teams-call) */
+          removedStatefulCall.push(teamsCallDeclaratify(call, this._context));
+        }
+      }
+    }
+
+    for (const externalCallsUpdatedListener of this._externalCallsUpdatedListeners) {
+      externalCallsUpdatedListener({ added: addedStatefulCall, removed: removedStatefulCall });
+    }
+  };
+
+  private setIncomingCallEnded = (incomingCallId: string, callEndReason: CallEndReason): void => {
+    const incomingCallSubscriber = this._incomingCallSubscribers.get(incomingCallId);
+    if (incomingCallSubscriber) {
+      incomingCallSubscriber.unsubscribe();
+      this._incomingCallSubscribers.delete(incomingCallId);
+    }
+    this._declarativeIncomingCalls.delete(incomingCallId);
+    this._context.setIncomingCallEnded(incomingCallId, callEndReason);
+  };
+
+  protected incomingCall = ({
+    incomingCall
+  }: {
+    incomingCall: IncomingCall | /* @conditional-compile-remove(teams-call) */ TeamsIncomingCall;
+  }): void => {
+    // Make sure to not subscribe to the incoming call if we are already subscribed to it.
+    if (!this._incomingCallSubscribers.has(incomingCall.id)) {
+      this._incomingCallSubscribers.set(
+        incomingCall.id,
+        new IncomingCallSubscriber(incomingCall, this.setIncomingCallEnded)
+      );
+    }
+    this._declarativeIncomingCalls.set(incomingCall.id, incomingCallDeclaratify(incomingCall, this._context));
+    this._context.setIncomingCall(convertSdkIncomingCallToDeclarativeIncomingCall(incomingCall));
+  };
+
+  protected addCall = (call: Call | /* @conditional-compile-remove(teams-call) */ TeamsCall): DeclarativeCallCommon => {
+    this._callSubscribers.get(call)?.unsubscribe();
+
+    // For API extentions we need to have the call in the state when we are subscribing as we may want to update the
+    // state during the subscription process in the subscriber so we add the call to state before subscribing.
+    this._context.setCall(convertSdkCallToDeclarativeCall(call));
+    this._callSubscribers.set(call, new CallSubscriber(call, this._context, this._internalContext));
+    return this.getOrCreateDeclarativeCall(call);
+  };
+
+  private getOrCreateDeclarativeCall = (
+    call: Call | /* @conditional-compile-remove(teams-call) */ TeamsCall
+  ): DeclarativeCallCommon => {
+    const declarativeCall = this._declarativeCalls.get(call);
+    if (declarativeCall) {
+      return declarativeCall;
+    }
+
+    /* @conditional-compile-remove(teams-call) */
+    if (!isACSCall(call)) {
+      const newDeclarativeCall = teamsCallDeclaratify(call, this._context);
+      this._declarativeCalls.set(call, newDeclarativeCall as DeclarativeTeamsCall);
+      return newDeclarativeCall;
+    }
+
+    const newDeclarativeCall = callDeclaratify(call, this._context);
+    this._declarativeCalls.set(call, newDeclarativeCall as DeclarativeCall);
+    return newDeclarativeCall;
+  };
+
+  public getCommon<
```

Comment:
`getCommon` should be `protected`.

Can this just be `protected get`?

Reply:
Answer: can't be `get` because you can't override functions that are being proxied.

Please add a comment to this function describing this.